lib/src/base_request.dart

```diff
@@ -99,7 +99,7 @@ abstract class BaseRequest {
   /// single-subscription to ensure that no data is dropped. They should also
   /// freeze any additional mutable fields they add that don't make sense to
   /// change after the request headers are sent.
-  ByteStream finalize() {
+  ByteStream? finalize() {
```

Comment:
Seems an interesting implementation detail that leaks around. Subclasses should *not* return `null` – but this does.

🤷

Reply:
Should this return an empty ByteStream or something?

Reply:
good idea!